### PR TITLE
[PAN-3063] Add method to fetch enclave url

### PR DESCRIPTION
**Pull request by iikirilov**
Adds interface method to give plugins knowledge of the enclave Pantheon is connected to

### CODE REVIEW

File: src/main/java/tech/pegasys/pantheon/plugin/services/PantheonConfiguration.java [hunk 1]
```diff
@@ -23,4 +25,12 @@
    * @return location of the storage in the file system of the client.
    */
   Path getStoragePath();
+
+  /**
+   * Url of the enclave Pantheon is connected to if privacy is enabled
+   *
+   * @return an optional containing the url of the enclave Pantheon is connected to, or empty if privacy
+   *     is not enabled
```
Comment (ajsutton) [inline src/main/java/tech/pegasys/pantheon/plugin/services/PantheonConfiguration.java]:
```suggestion
   *     is not enabled.
```

File: src/main/java/tech/pegasys/pantheon/plugin/services/PantheonConfiguration.java [hunk 2]
```diff
@@ -23,4 +25,12 @@
    * @return location of the storage in the file system of the client.
    */
   Path getStoragePath();
+
+  /**
+   * Url of the enclave Pantheon is connected to if privacy is enabled
```
Comment (ajsutton) [inline src/main/java/tech/pegasys/pantheon/plugin/services/PantheonConfiguration.java]:
```suggestion
   * Url of the enclave that stores private transaction data.
```

I'm not sure this is exactly right but trying to give a bit more flavour of what the enclave is actually used for.

**Comment by iikirilov** [comment]
> API looks good to me. The only concern is that at some point in the future we may need to support connecting to multiple Orions but since we don't know what the API would need to look like for that (Map vs List etc) we'll just have to deprecate this method if/when we do move to that model and provide an alternative that can handle that.

Multi-tenancy is far enough away to not require this for a while.

**Comment by ajsutton** [comment]
> Multi-tenancy is far enough away to not require this for a while.

Normally I'd agree as I'm a big fan of You Ain't Gonna Need it (YAGNI). Since the plugin APIs are intended to be public and highly stable we're trying to anticipate the future better so we can minimise the need to break it in future.  But you can't win 'em all.

**Pull request merged**